modules/hardware/allwinner/default.nix

```
@@ -94,14 +101,17 @@ in
     (mkIf cfg.allwinner-a64.enable {
       system.system = "aarch64-linux";
     })
-    (mkIf cfg.allwinner-h3.enable {
+    (mkIf (cfg.allwinner-h3.enable) {
       system.system = "armv7l-linux";
       Tow-Boot.config = [
         (helpers: with helpers; {
           CMD_POWEROFF = no;
         })
       ];
     })
+    (mkIf (cfg.allwinner-s3.enable) {
```

```suggestion
    (mkIf cfg.allwinner-s3.enable {
```

Not that it changes the result, but to keep the style consistent here.

* * *

Additionally, this should be placed after the `allwinner-h5` section, since this is ordered by name. (Not that it was documented to be,)